src/SQLite.jl

```diff
@@ -1,726 +1,25 @@
-function _Stmt(db::DB, sql::AbstractString)
-    handle = Ref{StmtHandle}()
-    sqliteprepare(db, sql, handle, Ref{StmtHandle}())
-    return _Stmt(handle[])
-end
-
-"""
-    SQLite.Stmt(db, sql) => SQL.Stmt
-
-Prepares an optimized internal representation of SQL statement in
-the context of the provided SQLite3 `db` and constructs the `SQLite.Stmt`
-Julia object that holds a reference to the prepared statement.
-
-*Note*: the `sql` statement is not actually executed, but only compiled
-(mainly for usage where the same statement is executed multiple times
-with different parameters bound as values).
-
-Internally `SQLite.Stmt` constructor creates the [`SQLite._Stmt`](@ref) object that is managed by `db`.
-`SQLite.Stmt` references the `SQLite._Stmt` by its unique id.
-
-The `SQLite.Stmt` will be automatically closed/shutdown when it goes out of scope
-(i.e. the end of the Julia session, end of a function call wherein it was created, etc.).
-One can also call `DBInterface.close!(stmt)` to immediately close it.
-
-All prepared statements of a given DB connection are also automatically closed when the
-DB is disconnected or when [`SQLite.finalize_statements!`](@ref) is explicitly called.
-"""
-mutable struct Stmt <: DBInterface.Statement
-    db::DB
-    id::_StmtId # id of _Stmt inside db (may refer to already closed connection)
-
-    function Stmt(db::DB, sql::AbstractString)
-        _stmt = _Stmt(db, sql)
-        id = (db.lastStmtId += 1)
-        stmt = new(db, id)
-        db.stmts[id] = _stmt # FIXME check for duplicate handle?
-        finalizer(_finalize, stmt)
-        return stmt
-    end
-end
-
-sqliteexception(db::DB, stmt::_Stmt) = sqliteexception(db.handle, stmt.handle)
-
-# check if the statement is ready (not finalized due to
-# _close(_Stmt) called and the statment handle removed from DB)
-isready(stmt::Stmt) = haskey(stmt.db.stmts, stmt.id)
-
-# get underlying _Stmt or nothing if not found
-_stmt_safe(stmt::Stmt) = get(stmt.db.stmts, stmt.id, nothing)
-
-# get underlying _Stmt or throw if not found
-@inline function _stmt(stmt::Stmt)
-    _st = _stmt_safe(stmt)
-    (_st === nothing) && throw(SQLiteException("Statement $(stmt.id) not found"))
-    return _st
-end
-
-# automatically finalizes prepared statement (_Stmt)
-# when no Stmt objects refer to it and removes
-# it from the db.stmts collection
-_finalize(stmt::Stmt) = DBInterface.close!(stmt)
-
-DBInterface.getconnection(stmt::Stmt) = stmt.db
-
-# explicitly close prepared statement
-function DBInterface.close!(stmt::Stmt)
-    _st = _stmt_safe(stmt)
-    if _st !== nothing
-        _close!(_st)
-        delete!(stmt.db.stmts, stmt.id) # remove the _Stmt
-    end
-    return stmt
-end
-
-sqliteprepare(db::DB, sql::AbstractString, stmt::Ref{StmtHandle}, null::Ref{StmtHandle}) =
-    @CHECK db sqlite3_prepare_v2(db.handle, sql, stmt, null)
-
+import Random
+import DBInterface
+using Serialization
+using WeakRefStrings
+using Tables
+
+include("capi.jl")
+import .C as C
```

Comment:
I think this can just be `import .C` since the `as` isn't renaming it at all. This is also causing test failures for Julia 1.3 (`as` didn't exist then). I'm also fine bumping the Julia version up to 1.6 though.